test/e2e/suite-endpoints/endpoints.go

```diff
@@ -21,17 +21,27 @@ import (
 
 	ginkgo "github.com/onsi/ginkgo/v2"
 	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/apache/apisix-ingress-controller/test/e2e/scaffold"
 )
 
+func ensureNumListUpstreamNodes(s *scaffold.Scaffold, upsNum int, upsNodesNum int) error {
```

Comment:
I think this function could be a public scaffold function instead of a package private one.